src/stream-tcp-private.h

```diff
@@ -64,6 +64,10 @@ typedef struct TcpSegment {
     uint32_t seq;
     RB_ENTRY(TcpSegment) __attribute__((__packed__)) rb;
     StreamingBufferSegment sbseg;
+    struct timeval ts;
```

Comment:
As I see you mentioned is that I think this should be optional. I wonder if we can reduce the overhead somehow if the feature is not used.